internal/checker/emitresolver.go

```diff
@@ -1,21 +1,586 @@
+		return false
+	}
+	if ast.NodeIsPresent(node.Body()) {
+		if ast.IsGetAccessorDeclaration(node) || ast.IsSetAccessorDeclaration(node) {
+			return false // Get or set accessors can never be overload implementations, but can have up to 2 signatures
+		}
+		r.checkerMu.Lock()
+		defer r.checkerMu.Unlock()
+		symbol := r.checker.getSymbolOfDeclaration(node)
+		signaturesOfSymbol := r.checker.getSignaturesOfSymbol(symbol)
+		// If this function body corresponds to function with multiple signature, it is implementation of overload
+		// e.g.: function foo(a: string): string;
+		//       function foo(a: number): number;
+		//       function foo(a: any) { // This is implementation of the overloads
+		//           return a;
+		//       }
+		return len(signaturesOfSymbol) > 1 ||
+			// If there is single signature for the symbol, it is overload if that signature isn't coming from the node
+			// e.g.: function foo(a: string): string;
+			//       function foo(a: any) { // This is implementation of the overloads
+			//           return a;
+			//       }
+			(len(signaturesOfSymbol) == 1 && signaturesOfSymbol[0].declaration != node)
+	}
+	return false
+}
+
+func (r *emitResolver) IsImportRequiredByAugmentation(decl *ast.ImportDeclaration) bool {
+	// node = r.emitContext.ParseNode(node)
+	if !ast.IsParseTreeNode(decl.AsNode()) {
+		return false
+	}
+	file := ast.GetSourceFileOfNode(decl.AsNode())
+	if file.Symbol == nil {
+		// script file
+		return false
+	}
+	importTarget := r.GetExternalModuleFileFromDeclaration(decl.AsNode())
+	if importTarget == nil {
+		return false
+	}
+	if importTarget == file {
+		return false
+	}
+	r.checkerMu.Lock()
+	defer r.checkerMu.Unlock()
+	exports := r.checker.getExportsOfModule(file.Symbol)
+	for s := range maps.Values(exports) {
+		merged := r.checker.getMergedSymbol(s)
+		if merged != s {
+			if len(merged.Declarations) > 0 {
+				for _, d := range merged.Declarations {
+					declFile := ast.GetSourceFileOfNode(d)
+					if declFile == importTarget {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func (r *emitResolver) RequiresAddingImplicitUndefined(declaration *ast.Node, symbol *ast.Symbol, enclosingDeclaration *ast.Node) bool {
+	// node = r.emitContext.ParseNode(node)
+	if !ast.IsParseTreeNode(declaration) {
+		return false
+	}
+	switch declaration.Kind {
+	case ast.KindPropertyDeclaration, ast.KindPropertySignature, ast.KindJSDocPropertyTag:
+		r.checkerMu.Lock()
+		defer r.checkerMu.Unlock()
+		if symbol == nil {
+			symbol = r.checker.getSymbolOfDeclaration(declaration)
+		}
+		t := r.checker.getTypeOfSymbol(symbol)
+		r.checker.mappedSymbolLinks.Has(symbol)
+		return !!((symbol.Flags&ast.SymbolFlagsProperty != 0) && (symbol.Flags&ast.SymbolFlagsOptional != 0) && isOptionalDeclaration(declaration) && r.checker.ReverseMappedSymbolLinks.Has(symbol) && r.checker.ReverseMappedSymbolLinks.Get(symbol).mappedType != nil && containsNonMissingUndefinedType(r.checker, t))
```

Comment:
This code here seems to be uncovered, which is interesting.

Reply:
AFAIK, the primary caller of this function is ID error generation, which isn't in yet. The current calls are only to handle some edge cases in declaration emit, the tested cases for which probably don't overlap with this specific condition. I know this line in strada was added to fix an ID bug, at least.